### ISSUE TITLE
FlatMap with a delayed inner error not cancelling other inner flowables 

### ISSUE THREAD

**Issue by Lukazoid** [akarnokd/Reactive4.NET#47]
Given the following example with delayError `false`:

```
async Task Main()
{
	var flowable = Flowable.FromEnumerable(Enumerable.Range(1, 5));

	try
	{
		await flowable.FlatMap(x => Flowable.Defer(() =>
		{
			Console.WriteLine($"{x}: subscribed");
			if (x == 1)
				return Flowable.Error<long>(new Exception("My Test Exception")).Delay(TimeSpan.FromMilliseconds(100));

			return Flowable.Interval(TimeSpan.FromSeconds(1))
				.DoOnNext(_ => Console.WriteLine($"{x}: next"))
				.DoOnCancel(() => Console.WriteLine($"{x}: canceled"));
		}), int.MaxValue, 1024, false)
		.IgnoreElementsTask(CancellationToken.None);
	}
	finally
	{
		Console.WriteLine("Program Ended");
	}

	Console.ReadKey();
}
```

The other inner flowables never get canceled and their `next` messages continue to be printed to console.

I believe due to delayError false, as soon as one inner flowable reports an error then all the others should be getting canceled? (code from `FlatMapMainSubscriber.InnerError`)

Also of note, if I reduce bufferSize from 1024 to 2, then it successfully stops the inner IFlowables after each has emitted 2 items, however the "canceled" message is never printed.

**Comment by akarnokd** [comment]
Thanks for the report. Indeed it seems there is an extra check that doesn't appear in the RxJava implementation that prevents the cancellation. I'll fix it shortly.

**Comment by akarnokd** [comment]
Fixed in **0.2.25**.
